test/crypto.js

```diff
@@ -15,66 +15,92 @@ describe('MasqCommon crypto', function () {
+      const MK = await MasqCommon.crypto.decryptMasterKey(passphrase, protectedMK)
+      const cryptokey = await MasqCommon.crypto.importKey(MK)
+      const data = { 'hello': 'world' }
+      const enc = await MasqCommon.crypto.encrypt(cryptokey, data)
+      chai.assert.exists(enc.iv, 'iv must exist')
+      chai.assert.exists(enc.ciphertext, 'ciphertext must exist')
+
+      // Just to be sure that everything is working well.
+      const sameMK = await MasqCommon.crypto.decryptMasterKey(passphrase, protectedMK)
+      const sameCryptokey = await MasqCommon.crypto.importKey(sameMK)
+      const dec = await MasqCommon.crypto.decrypt(sameCryptokey, enc)
+      chai.assert.deepEqual(dec, data, 'Must be the same')
     })
 
-    it('Should reject if the any property of hashedPassphrase is missing or empty', async () => {
+    it('Should reject if the given passphrase is NOT the same as the stored one', async () => {
+      let err = { type: '_ERROR_NOT_THROWN_' }
+      try {
+        const protectedMK = await MasqCommon.crypto.genEncryptedMasterKey(passphrase)
+        await MasqCommon.crypto.decryptMasterKey(passphrase + 'modifed', protectedMK)
+      } catch (error) {
+        err = error
+      }
+      chai.assert.equal(err.type, MasqCommon.errors.ERRORS.WRONGPASSPHRASE, 'Reject if wrong passphrase')
+    })
+
+    it('Should reject if the any property of protectedMK is missing or empty', async () => {
       let err = '_ERROR_NOT_THROWN_'
       try {
-        await MasqCommon.crypto.checkPassphrase('secretPassphraseCandidate', {})
+        await MasqCommon.crypto.decryptMasterKey('secretPassphraseCandidate', {})
       } catch (error) {
         err = error.name
       }
       chai.assert.equal(err, MasqCommon.errors.ERRORS.WRONGPARAMETER, 'A requried property is missing')
     })
 
-    it('The salt and storedHash must be different for two consecutive call to derivePassphrase even with the same passphrase', async () => {
+    it('The salt and protectedMK must be different for two consecutive call to genEncryptedMasterKey even with the same passphrase', async () => {
       const passphrase = 'secret'
-      const hashedPassphrase1 = await MasqCommon.crypto.derivePassphrase(passphrase)
-      const hashedPassphrase2 = await MasqCommon.crypto.derivePassphrase(passphrase)
-      chai.assert.equal(hashedPassphrase1.salt === hashedPassphrase2.salt, false, 'Two different salt')
-      chai.assert.equal(hashedPassphrase1.storedHash === hashedPassphrase2.storedHash, false, 'Two different salt')
+      const protectedMK1 = await MasqCommon.crypto.genEncryptedMasterKey(passphrase)
+      const protectedMK2 = await MasqCommon.crypto.genEncryptedMasterKey(passphrase)
+      chai.assert.equal(protectedMK1.derivationParams.salt === protectedMK2.derivationParams.salt, false, 'Two different salt')
+      chai.assert.equal(protectedMK1.encryptedMasterKey.iv === protectedMK2.encryptedMasterKey.iv, false, 'Two different iv')
```

Comment:
You should use `notStrictEqual` instead https://www.chaijs.com/api/assert/#method_notstrictequal

Reply:
ok done